lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2466,7 +2507,11 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
               PBD->getInitContext(i));
           if (initContext) {
             TypeChecker::contextualizeInitializer(initContext, init);
-            checkInitializerActorIsolation(initContext, init);
+            if (auto *singleVar = PBD->getSingleVar()) {
+              (void)singleVar->getInitializerIsolation();
+            } else {
+              computeRequiredIsolation(initContext, init);
+            }
```

Comment:
Why not simplify `computeRequiredIsolation` because this is already happening in the loop that goes over the entries in the pattern binding?

Reply:
Oh my got a wrote a giant comment and forgot to add it :/

Reply:
This means that a single-variable pattern is going to be cached but multi-variable one is not going to. I want to suggest adding `Pattern *` as a parameter to `computeRequiredIsolation` so that after isolation is computed it could collect all of the variables involved and cache the result for each, this way we can simply use `computeRequiredIsolation` here and in other places without this re-computation caveat.

Another way might be to compute required isolation one for anchoring var decl after the loop is done.

Reply:
Fixed this to always go through the request and added a `PatternBindingDecl::getInitializerIsolation(unsigned i)` convenience to get the initializer isolation for each pattern binding entry.